temoa/core/config.py

```
@@ -157,6 +154,42 @@ def __init__(
+            import pyomo.environ as pyo
+
+            solver = pyo.SolverFactory(solver_name)
+            if solver.available():
+                # For Python-based solvers like appsi_highs
+                if solver_name.startswith('appsi_'):
+                    return True, 'available via Pyomo (Python package)'
+                # For other solvers, try to get the executable path
+                try:
+                    exec_path = solver.executable()
+                    return True, exec_path.name if hasattr(exec_path, 'name') else str(exec_path)
+                except Exception:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)